components/SwipeRow.js

```diff
@@ -161,17 +170,35 @@ class SwipeRow extends Component {
 
         if (this.props.preview && !this.ranPreview) {
             this.ranPreview = true;
-            const previewOpenValue =
-                this.props.previewOpenValue || this.props.rightOpenValue * 0.5;
-            this.getPreviewAnimation(
-                previewOpenValue,
-                this.props.previewOpenDelay
-            ).start(() => {
-                this.getPreviewAnimation(0, PREVIEW_CLOSE_DELAY).start();
-            });
+            if (this.props.previewRepeat) {
+                const totalRepeatDelay =
+                    (this.props.previewDuration +
+                        this.props.previewOpenDelay +
+                        PREVIEW_CLOSE_DELAY) *
+                        2.5 +
+                    this.props.previewRepeatDelay;
```

Comment:
Can you please explain this calculation? In testing this I'm noticing the first preview takes a very long time to start. Using the defaults the initial delay would be (300 + 700 + 300) * 2.5 =  3250 ms or 3.25 seconds which seems too long to me. What do you think?

Reply:
yeah there was a few things wrong with this, ill explain in response with commit